src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -2121,6 +2162,11 @@ public void onWakeUp()
             setLatestStatus(new TextComponentTranslation("com.minecolonies.coremod.status.working"));
             this.getWorkBuilding().onWakeUp();
         }
+
+        if(this.getHomeBuilding() != null)
+        {
+            this.getHomeBuilding().onWakeUp();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getHomeBuilding" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)